File: server/Instagram.API/Domain/Models/Post.cs
using Newtonsoft.Json;

namespace Instagram.API.Domain.Models;

public class Post : BaseAuditableEntity
{
    public Guid PostId { get; set; }
    public Guid UserId { get; set; }
    public string Caption { get; set; }
    public string MediaPath { get; set; }
    public IList<Like> Likes { get; private set; } = new List<Like>();
    public IList<Comment> Comments { get; private set; } = new List<Comment>();
    [JsonIgnore]
    public User User { get; set; }
}

File: client/src/core/models/Post.ts
export interface Comment {
    commentId?: string;
    postId?: string;
    userId?: string;
    commentText?: string;
    commentBy?: string;
    created?: string;
    lastModified?: string;
}

export interface Post {
    postId: string;
    userId?: string;
    username?: string;
    created?: string;
    lastModified?: string;
    caption?: string;
    mediaPath?: string;
    likes: number;
    comments: Comment[];
    isLiked?: boolean;
    isFollow?: boolean;
}

export interface PostDetail {
    postId: string;
    userId: string;
    username: string;
    mediaPath: string;
    likes: number;
    caption: string;
    isLiked: boolean;
    comments: Comment[];
    lastModified?: string;
}
